core/dbt/adapters/base/connections.py

```diff
@@ -34,6 +48,9 @@
 )
 from dbt import flags
 
+SleepTime = Union[int, float]  # As taken by time.sleep.
```

Comment:
alternatively, is there even a use case for specifying seconds more specifically that integer granularity?

Reply:
Not that I can think of. I opted for the union as that's what `time.sleep` can take, so we can leave it up to adapter maintainers to use whatever they want.

I would be okay with restricting the type hint though, as I don't think it's very critical either way.

Reply:
@dataders there's also not a use case to restrict the input values to just ints. since float is (roughly) a superset of int, I'd rather just choose the bigger type. The union is totally functional, it's just add a level of indirection I don't think is totally necessary.

Reply:
either option of `float` or `SleepTime = Union[int, float]` is fine with me.